### ISSUE TITLE
day9 part 2 recursive call

### ISSUE THREAD

**Issue by chrismurrph** [bhauman/advent-of-clojure-2016#2]
Function 'part2' calls like this: `(part22 part)`. But it should be calling itself like this: `(part2 part)`.

**Comment by chrismurrph** [comment]
Also in the `part1` function shouldn't `(starts-with-directive? d)` instead be `(starts-with-directive? data)`?

**Comment by bhauman** [comment]
oh my god yes :)

**Comment by bhauman** [comment]
Strange and confusing that it works with that crazy bug??

**Comment by bhauman** [comment]
@chrismurrph I'm now going through the rest of the solutions to double check everything :)

BTW I'm really glad you are going through these and finding it helpful.

**Comment by chrismurrph** [comment]
When I saw that bug I couldn't help but think that `starts-with-directive?` was unnecessary. And so I commented it out and the code for both parts still comes up with the right answers:
https://github.com/chrismurrph/advent-of-code/blob/master/dev/advent/nine.clj#L244
Even although the purpose of `starts-with-directive?` was to avoid doing a big lazy-seq -> string every time, not avoiding it didn't exactly have me twiddling my thumbs for moments on end.

So far my answers have all been long winded compared to yours. I couldn't think of a better way to become more proficient. Nine was good for those regex functions I would never have otherwise discovered. 

Whilst here, maybe in 10 `(u/to-int (first (re-seq #"\d+" l)))` could be changed to `(u/to-int (re-find #"\d+" l))`?
